runtime/gc_glue_java/ScavengerRootClearer.cpp

```diff
@@ -238,12 +238,14 @@ MM_ScavengerRootClearer::scavengeContinuationObjects(MM_EnvironmentStandard *env
 							gcEnv->_scavengerJavaStats._continuationCandidates += 1;
 
 							MM_ForwardedHeader forwardedHeader(object, compressed);
-							if (!forwardedHeader.isForwardedPointer()) {
-								Assert_GC_true_with_message2(env, _scavenger->isObjectInEvacuateMemory(object), "Continuation object  %p should be a dead object, forwardedHeader=%p\n", object, &forwardedHeader);
+							omrobjectptr_t forwardedPtr = object;
+							if (forwardedHeader.isForwardedPointer()) {
+								forwardedPtr = forwardedHeader.getForwardedObject();
+							}
+							if (!forwardedHeader.isForwardedPointer() || VM_VMHelpers::isContinuationFinished((J9VMThread *)env->getLanguageVMThread(), forwardedPtr)) {
 								gcEnv->_scavengerJavaStats._continuationCleared += 1;
-								_extensions->releaseNativesForContinuationObject(env, object);
+								_extensions->releaseNativesForContinuationObject(env, forwardedPtr);
 							} else {
-								omrobjectptr_t forwardedPtr = forwardedHeader.getForwardedObject();
 								Assert_GC_true_with_message(env, NULL != forwardedPtr, "Continuation object  %p should be forwarded\n", object);
```

Comment:
this assert might be too late - we might have already dereferenced forwardedPtr as part of isContinuationFinished, perhaps move it just after getForwardedObject?